### DISCUSSION TITLE
function explain, help! getAudioFileDuration()

### DISCUSSION THREAD

**Discussion by MIKHANYA** [schreibfaul1/ESP32-audioI2S#298] [Q&A] [answered]
Need help, please explain how to use this function`getAudioFileDuration()` I'm trying to read file duration on SD card.
 Code that posted in wiki page, https://github.com/schreibfaul1/ESP32-audioI2S/wiki#what-else-is-there what is ticker ? It doesn't compile with includes like
```
#include "Arduino.h"
#include "WiFiMulti.h"
#include "Audio.h"
#include "SPI.h"
#include "SD.h"
#include "FS.h"
```

````
Ticker ticker;
void setup()  {
	...
	ticker.attach(1, tcr1s);
	...
}

void tcr1s(){
    uint32_t act=audio.getAudioCurrentTime();
    uint32_t afd=audio.getAudioFileDuration();
    uint32_t pos =audio.getFilePos();
    log_i("pos =%i", pos);
    log_i("audioTime: %i:%02d - duration: %i:%02d", (act/60), (act%60) , (afd/60), (afd%60));
}
````

**Comment by schreibfaul1** [comment, marked as answer]
Ticker is a class in Ticker.h that periodically calls an interrupt via a callback function.
Please try `#include "Ticker.h"`, I think that's missing in the example
